Move and refactor integration-cli/registry to internal/test

vdemeester opened the pull request:
- Move the code from `integration-cli` to `internal/test`.
- Use `testingT` and `assert` when creating the registry.

:lion: 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

cpuguy83 commented on internal/test/registry/registry.go:
```diff
@@ -80,39 +94,51 @@ http:
+}
+
+// WaitFor waits for the registry to be ready to serve requests (or fail after a while)
+func (r *V2) WaitFor(t testingT) {
```
Nit, `WaitReady` seems more appropriate here.

cpuguy83 reviewed (approved):
LGTM

cpuguy83 commented:
```
14:14:22 internal/test/registry/ops.go:22:6:warning: func name will be used as registry.RegistryURL by other packages, and that stutters; consider calling this URL (golint)
```

yongtang reviewed (approved):
LGTM

vdemeester committed:
Move and refactor integration-cli/registry to internal/test

- Move the code from `integration-cli` to `internal/test`.
- Use `testingT` and `assert` when creating the registry.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

thaJeztah reviewed (approved):
LGTM, thanks!
